handshake_messages.go

```diff
@@ -407,6 +412,11 @@ func (m *clientHelloMsg) marshal() []byte {
 		z[1] = byte(extensionEarlyData)
 		z = z[4:]
 	}
+	if m.delegatedCredential {
+		z[0] = byte(extensionDelegatedCredential >> 8)
+		z[1] = byte(extensionDelegatedCredential & 0xff)
```

Comment:
the two times when this is done before, there's no & 0xff

Reply:
Yes, because the other extension numbers are less than 256 and so fit in a single byte; ours doesn't. For example, `byte(0xffff)` will give you an error, but `byte(0xff)` will not.

Reply:
https://play.golang.org/p/N5GsqbuKr91

Reply:
Hmmm ... I upgraded to 1.10.3 and got the same error. Have you tried running this on your machine?

Reply:
I see... it's doing compile-time overflow checking because it's a const, not a var. And the others don't need &0xff because they're genuinely less than 2^8. I'm sorry, disregard this

Reply:
Yeah, I just figured out the same :)

https://play.golang.org/p/jpFT2CDPV-i

Reply:
Done.